pybamm/models/submodels/convection/transverse/__init__.py

```diff
@@ -1,4 +1,30 @@
-from .base_transverse_convection import BaseTransverseModel
-from .no_convection import NoConvection
-from .uniform_convection import Uniform
-from .full_convection import Full
+import lazy_loader
+
+
+__getattr__, __dir__, __all__ = lazy_loader.attach(
+    __name__,
+    submodules={
+        'base_transverse_convection',
+        'full_convection',
+        'no_convection',
+        'uniform_convection',
+    },
+    submod_attrs={
+        'base_transverse_convection': [
+            'BaseTransverseModel',
+        ],
+        'full_convection': [
+            'Full',
+        ],
+        'no_convection': [
+            'NoConvection',
+        ],
+        'uniform_convection': [
+            'Uniform',
+        ],
+    },
+)
+
+__all__ = ['BaseTransverseModel', 'Full', 'NoConvection', 'Uniform',
+           'base_transverse_convection', 'full_convection', 'no_convection',
+           'uniform_convection']
```

Comment:
Adding all of this to every init seems like a lot. How much time does this really save?

Reply:
Seems like  lot of code and complexity if it is only shaving off a little overhead

Reply:
Yes it seems like a lot but again it is generated using `mkinit` and will be maintained using same if implemented. 
As for time, Usually `import pybamm` can take you 15 to 20 seconds but after lazy-loading it barely takes a second in most of the cases.

Reply:
I have never seen it take that long to import. Maybe half a second for me